assignments/lambda-classes.js

```diff
@@ -1 +1,88 @@
 // CODE here for your Lambda Classes
+//base class
+class Person {
+  constructor(attr) {
+    this.name = attr.name;
+    this.age = attr.age;
+    this.location = attr.location;
+  }
+
+  speak() {
+    console.log(`Hello my name is ${this.name}, I am from ${this.location}`);
+  }
+}
+class Instructor extends Person {
+  constructor(attrs) {
+    super(attrs);
+    this.specialty = attrs.specialty;
+    this.favLanguage = attrs.favLanguage;
+    this.catchPhrase = attrs.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning ${subject}`);
+  }
+
+  grade(student, subject) {
+    console.log(`${this.name} recieves a perfect score on ${subject}`);
+  }
+}
+class Student extends Instructor {
+  constructor(attr) {
+    super(attr);
+    this.previousBackground = attr.previousBackground;
+    this.className = attr.className;
+    this.favSubjects = attr.favSubjects;
+  }
+
+  listSubjects(cb) {
+    setTimeout(function() {
```

Comment:
sorry if my standup name randomizer caused confusion for you here. there is no need to delay your function here. The set timeout is mainly used to simulate asynchronous calls to a server (totally fine if you don't know what that means. You will learn all about it in a few months) What you want to do here is console log a student's favorite subjects.